src/arrays.js

```diff
@@ -1,61 +1,157 @@
 };
 
 const addToArray2 = (element, array) => {
-  // your code here
+  if (element !== '') {
+    const array2 = array.concat(element);
+    return array2;
+  }
+  return array;
 };
 
 const removeNthElement = (index, array) => {
-  // your code here
+  return array.splice(index, 1);
 };
 
 const numbersToStrings = (numbers) => {
-  // your code here
+  // const numberStringArray = [];
+  // for (let i = 0; i < numbers.length; i++) {
+  //   numberStringArray.push(numbers[i].toString());
+  // }
+  // return numberStringArray;
+
+  function turnIntoString(numberasnumber) {
+    return numberasnumber.toString();
+  }
+  return numbers.map(turnIntoString);
 };
 
 const uppercaseWordsInArray = (strings) => {
-  // your code here
+  // const upperCaseArray = [];
+  // for (let i = 0; i < strings.length; i++) {
+  //   upperCaseArray.push(strings[i].toUpperCase());
+  // }
+  // return upperCaseArray;
+
+  function toUpperCase(lowerCaseString) {
+    return lowerCaseString.toUpperCase();
+  }
+  return strings.map(toUpperCase);
 };
 
 const reverseWordsInArray = (strings) => {
-  // your code here
+  const reversedStrings = [];
+  for (let i = 0; i < strings.length; i++) {
+    const currentString = strings[i];
+    let reverseString = '';
+    for (let j = currentString.length - 1; j >= 0; j--) {
+      reverseString += currentString[j];
+    }
+    reversedStrings.push(reverseString);
+  }
+  return reversedStrings;
 };
 
 const onlyEven = (numbers) => {
-  // your code here
+  const evens = [];
+  return numbers.filter((element) => {
+    if (element % 2 !== 0) {
+      return (element[0]);
+    }
+    return evens;
+
+    
+  });
 };
 
 const removeNthElement2 = (index, array) => {
-  // your code here
+  const newArray = array.slice(0, index).concat(array.slice(index + 1, array.length));
+  return newArray;
 };
 
 const elementsStartingWithAVowel = (strings) => {
-  // your code here
+  // const vowelElements = [];
+  // for (let i = 0; i < strings.length; i++) {
+  //   const currentString = strings[i];
+  //   const lowerString = currentString.toLowerCase();
+  //   if (lowerString[0] === 'a' || lowerString[0] === 'e' || lowerString[0] === 'i' || lowerString[0] === 'o' || lowerString[0] === 'u') {
+  //     vowelElements.push(currentString);
+  //   }
+  // }
+  // return vowelElements;
+
+  return strings.filter((element) => {
+    const lowerString = element.toLowerCase();
+    return lowerString[0] === 'a' || lowerString[0] === 'e' || lowerString[0] === 'i' || lowerString[0] === 'o' || lowerString[0] === 'u';
+  });
 };
 
 const removeSpaces = (string) => {
-  // your code here
+  const noSpaceString = string.replace(/\s/g, '');
+  return noSpaceString;
 };
 
 const sumNumbers = (numbers) => {
-  // your code here
+  // let total = 0;
+  // for (let i = 0; i < numbers.length; i++) {
+  //   total += numbers[i];
+  // }
+  // return total;
+
+  function getSum(a, b) {
+    return a + b;
+  }
+  return numbers.reduce(getSum);
```

Comment:
you need to provide a starting value to your reducer!

` return numbers.reduce(getSum, 0);`